private/buf/bufwire/file_lister.go

```diff
@@ -61,6 +77,74 @@ func (e *fileLister) ListFiles(
 	container app.EnvStdinContainer,
 	ref buffetch.Ref,
 	configOverride string,
+	includeImports bool,
+) ([]bufmoduleref.FileInfo, []bufanalysis.FileAnnotation, error) {
+	if includeImports {
+		// To get imports, we need to build an image so we keep this flow separate and
+		// re-use the logic in imageConfigReader.
+		return e.listFilesWithImports(
+			ctx,
+			container,
+			ref,
+			configOverride,
+		)
+	}
+	// We don't need to build in the withoutImports flow, so we just completely separate the logic
+	// to make sure the common case is as quick as it can be.
+	fileInfos, err := e.listFilesWithoutImports(
+		ctx,
+		container,
+		ref,
+		configOverride,
+	)
+	return fileInfos, nil, err
+}
+
+func (e *fileLister) listFilesWithImports(
+	ctx context.Context,
+	container app.EnvStdinContainer,
+	ref buffetch.Ref,
+	configOverride string,
+) ([]bufmoduleref.FileInfo, []bufanalysis.FileAnnotation, error) {
+	imageConfigs, fileAnnotations, err := e.imageConfigReader.GetImageConfigs(
+		ctx,
+		container,
+		ref,
+		configOverride,
+		nil,
+		false,
+		true,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(fileAnnotations) > 0 {
+		return nil, fileAnnotations, nil
+	}
+	images := make([]bufimage.Image, len(imageConfigs))
+	for i, imageConfig := range imageConfigs {
+		images[i] = imageConfig.Image()
+	}
+	image, err := bufimage.MergeImages(images...)
+	if err != nil {
+		return nil, nil, err
+	}
+	imageFiles := image.Files()
+	fileInfos := make([]bufmoduleref.FileInfo, len(imageFiles))
+	for i, imageFile := range imageFiles {
+		fileInfos[i] = imageFile
+	}
+	// The image is ordered in DAG order, which isn't consistent with what we do
+	// in the withoutImports flow and is a little annoying for a user.
+	bufmoduleref.SortFileInfos(fileInfos)
```

Comment:
Maybe we should do it in DAG order anyways? It is more consistent with other outputs I think, I don't know.

Reply:
I think I prefer the alpha-sorted order, seems easier to reason about, from a human readability standpoint. But if we're expecting this to be used as inputs to another process (which seems pretty likely), I think either is fine. But it doesn't seem too costly to sort, so it might be beneficial to have that just for the use-cases where users are inspecting the output.

Reply:
As long as it's deterministic, we should feel empowered to change the ordering later if need be.

At first glance, it seems best to keep this consistent with the `withoutImports` flow.